From a82d202af20f3bd10a426331c19ada6548b97f0f Mon Sep 17 00:00:00 2001
From: Thierry FOURNIER <tfournier@exceliance.fr>
Date: Thu, 5 Dec 2013 11:59:55 +0100
Subject: [PATCH 191/203] MINOR: map: Cleanup the initialisation of map
 descriptors.

The association between struct map_descriptor and its struct
map_reference must be done in the function map_create_descriptor().
---
 src/map.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/map.c b/src/map.c
index 81c61eb..2ed9f3a 100644
--- a/src/map.c
+++ b/src/map.c
@@ -168,6 +168,7 @@ static struct map_descriptor *map_create_descriptor(struct map_reference *ref,
 		return NULL;
 
 	desc->conv = conv;
+	desc->ref = ref;
 
 	LIST_ADDQ(&ref->maps, &desc->list);
 
@@ -366,7 +367,6 @@ static int sample_load_map(struct arg *arg, struct sample_conv *conv, char **err
 		return 0;
 	}
 
-	desc->ref = ref;
 	pattern_init_expr(&desc->pat);
 
 	/* set the match method */
-- 
1.8.1.5

